### PR TITLE
*: update util/tracing and remove repeated code

### LINKED ISSUE #40809: remove repeated tracing code
## Enhancement

There are many repeated code like this:

https://github.com/pingcap/tidb/blob/4686338d0eef4135b41a7f4a5a70ee80ff267040/session/session.go#L1540-L1544

1. The `opentracing` code block is used in the `trace` SQL command 

```
trace select * from t limit 1
```

2. The `trace.StartRegion()` is used by the `go tool trace`

Both of them need to insert some code snippet to support the tracing utility.
And they are very similiar.

It's better to integrate those two into one, so as to reduce repeated code.

When tracing is disabled (default option), there should be no performance impact.
Thus, the more detailed information, the better.

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #40809

Problem Summary:

In the past, we insert some trace point for opentracing and for runtime.trace.
The code is repeated.

### What is changed and how it works?

Wrap the code in a package to reduce repeated code.
The runtime.trace code is more detailed now.

After this change, we can see more useful information throw `go tool trace` and answer many questions like:

- Why this query is slow (P99)? 
- What happens when the tso latency is so high?
- ...

image

image

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [X] Manual test (add detailed scripts or steps below)

Run some tpch workload and then

```
curl 'http://127.0.0.1:10080/debug/pprof/trace?seconds=3' > trace.out
go tool trace trace.out
```

Navigate the html result in the browser, see those pictures above.

- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: session/bootstrap.go [hunk 1]
```diff
@@ -964,6 +964,9 @@ func getTiDBVar(s Session, name string) (sVal string, isNull bool, e error) {
 // For example, add new system variables into mysql.global_variables table.
 func upgrade(s Session) {
 	ver, err := getBootstrapVersion(s)
+	if err != nil {
+		fmt.Println("ErrorStack:", errors.ErrorStack(err))
```
Comment (hawkingrei) [inline session/bootstrap.go:968]: Should use ```log.Error``` ? It looks like a debug code.

File: tidb-server/main.go [hunk 2]
```diff
@@ -331,6 +331,9 @@ func createStoreAndDomain(keyspaceName string) (kv.Storage, *domain.Domain) {
 	terror.MustNil(err)
 	// Bootstrap a session to load information schema.
 	dom, err := session.BootstrapSession(storage)
+	if err != nil {
+		fmt.Println("ErrorStack", errors.ErrorStack(err))
```
Comment (hawkingrei) [inline tidb-server/main.go:335]: ditto

**Comment by hawkingrei** [comment]
Please run ```make bazel_prepare```.

File: session/bootstrap.go (hunk 1 above)
Comment (tiancaiamao) [reply session/bootstrap.go:968]: The debugging log added mistakenly, I'll remove this

**Review by hawkingrei — APPROVED** [review]

File: executor/adapter.go [hunk 3]
```diff
@@ -283,12 +283,12 @@ func (a *ExecStmt) GetStmtNode() ast.StmtNode {
...
 // PointGet short path for point exec directly from plan, keep only necessary steps
 func (a *ExecStmt) PointGet(ctx context.Context) (*recordSet, error) {
-	if span := opentracing.SpanFromContext(ctx); span != nil && span.Tracer() != nil {
-		span1 := span.Tracer().StartSpan("ExecStmt.PointGet", opentracing.ChildOf(span.Context()))
-		span1.LogKV("sql", a.OriginText())
-		defer span1.Finish()
-		ctx = opentracing.ContextWithSpan(ctx, span1)
+	r, ctx := tracing.StartRegionEx(ctx, "ExecStmt.PointGet")
+	defer r.End()
+	if r.Span != nil {
```
Comment (XuHuaiyu) [inline executor/adapter.go:288]: why check nil here?

**Review by XuHuaiyu — APPROVED** [review]

File: executor/adapter.go (hunk 3 above)
Comment (tiancaiamao) [reply executor/adapter.go:288]: If opentracing config is not enabled or in the `trace xxx` statement, `r.Span`  would be nil and the following `r.Span.LogKV` would panic.

Comment (tiancaiamao) [reply executor/adapter.go:288]: So ...  to make the code robust

**Pull request merged**